common/src/main/scala/react/resizeDetector/hooks.scala

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package react.resizeDetector
+
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.{ facade => Raw }
+import org.scalajs.dom.html
+import react.common._
+import react.resizeDetector.ResizeDetector._
+
+import scala.annotation.nowarn
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+import scala.scalajs.js.|
+
+@js.native
+protected trait ReactResizeDetectorDimensions extends js.Object {
+  val height: js.UndefOr[Raw.JsNumber]
+  val width: js.UndefOr[Raw.JsNumber]
+}
+
+@js.native
+trait UseResizeDetectorReturnJS extends ReactResizeDetectorDimensions {
+  val ref: Raw.React.RefFn[html.Element]
+}
+
+sealed trait UseResizeDetectorReturn {
+  val height: Option[Int]
+  val width: Option[Int]
+  val ref: Ref.Simple[html.Element]
+}
+
+object UseResizeDetectorReturn {
+  def fromJS(r: UseResizeDetectorReturnJS): UseResizeDetectorReturn = new UseResizeDetectorReturn {
+    val height = r.height.toOption.map(_.toInt)
+    val width  = r.width.toOption.map(_.toInt)
+    val ref    =
+      Ref.fromJs(r.ref.asInstanceOf[facade.React.RefHandle[html.Element | Null]])
+  }
+
+  implicit val reusabilityUseResizeDetectorReturn: Reusability[UseResizeDetectorReturn] =
+    Reusability.by(x => (x.height, x.width))
+}
+
+@js.native
+protected trait UseResizeDetectorProps extends Props {
+  var targetRef: Raw.React.RefFn[html.Element]
+}
+
+object UseResizeDetectorProps {
+
+  def apply(
+    onResize:        js.UndefOr[(Int, Int) => Callback] = js.undefined,
+    handleHeight:    js.UndefOr[Boolean] = js.undefined,
+    handleWidth:     js.UndefOr[Boolean] = js.undefined,
+    skipOnMount:     js.UndefOr[Boolean] = js.undefined,
+    refreshMode:     js.UndefOr[RefreshMode] = js.undefined,
+    refreshRate:     js.UndefOr[Int] = js.undefined,
+    refreshOptions:  js.UndefOr[RefreshOptions] = js.undefined,
+    observerOptions: js.UndefOr[ObserverOptions] = js.undefined
+  ): UseResizeDetectorProps = {
+    val p = (new js.Object).asInstanceOf[UseResizeDetectorProps]
+    onResize.foreach(v =>
+      p.onResize = { case (x: Raw.JsNumber, y: Raw.JsNumber) =>
+        v(x.toInt, y.toInt).runNow()
+      }: js.Function2[
+        Raw.JsNumber,
+        Raw.JsNumber,
+        Unit
+      ]
+    )
+    handleHeight.foreach(v => p.handleHeight = v)
+    handleWidth.foreach(v => p.handleWidth = v)
+    skipOnMount.foreach(v => p.skipOnMount = v)
+    refreshMode.toJs.foreach(v => p.refreshMode = v)
+    refreshRate.foreach(v => p.refreshRate = v)
+    refreshOptions.foreach(v => p.refreshOptions = v)
+    observerOptions.foreach(v => p.observerOptions = v)
+    p
+  }
+}
+
+object HooksApiExt {
+
+  object mod {
+    @JSImport("react-resize-detector", JSImport.Namespace)
+    @js.native
+    @nowarn
```

Comment:
Why is `@nowarn` needed?

Reply:
If `base` is not to be referenced from outside maybe we can make it `private`? And define it directly as `js.Dynamic`.

Reply:
Somehow it detects that is not used. If I make it private it should do fine I thin